.github/workflows/nix_flake_update.yml

```
@@ -0,0 +1,26 @@
+---
+name: Nix Flake Update
+
+on:  # yamllint disable-line rule:truthy
+  workflow_dispatch:
+
+  schedule:
+    - cron: "0 0 1 * *"
+
+jobs:
+  nix-flake-update:
+    runs-on: ubuntu-22.04
+
+    permissions:
+      contents: write
+      pull-requests: write
+
+    steps:
+      - uses: actions/checkout@v4
+      - uses: DeterminateSystems/nix-installer-action@v16
+      - uses: DeterminateSystems/magic-nix-cache-action@v8
```

magic nix cache doesn't work anymore unless on enterprise server, so please switch to flakehub.
https://determinate.systems/posts/magic-nix-cache-free-tier-eol/

```suggestion
      - uses: DeterminateSystems/flakehub-cache-actio@main
```

---

we've been using cachix for are other actions, so this should probably also use it.

---

Yes, this is just old code since the PR has not been updated in a while.